File: server/DocumentsKM/Models/ConstructionBolt.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Высокопрочный болт
    public class ConstructionBolt
    {
        [Key]
        public Int16 Id { get; set; }

        // Вид конструкции
        [Required]
        [ForeignKey("ConstructionId")]
        public virtual Construction Construction { get; set; }

        // Диаметр
        [Required]
        [ForeignKey("DiameterId")]
        public virtual BoltDiameter Diameter { get; set; }

        // Толщина пакета
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16 Packet { get; set; }

        // Количество болтов
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16 Num { get; set; }

        // Количество гаек
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16 NutNum { get; set; }

        // Количество шайб
        [Required]
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16 WasherNum { get; set; }
    }
}

File: client/src/model/ConstructionBolt.ts
import BoltDiameter from './BoltDiameter'

interface IConstructionBolt {
	id: number
	diameter: BoltDiameter
	packet: number
	num: number
	nutNum: number
	washerNum: number
}

class ConstructionBolt {
	id: number
	diameter: BoltDiameter
	packet: number
	num: number
	nutNum: number
	washerNum: number

	constructor(obj?: IConstructionBolt) {
		this.id = (obj && obj.id) || 0
		this.diameter = (obj && obj.diameter) || null
		this.packet = (obj && obj.packet) || 0
		this.num = (obj && obj.num) || 0
		this.nutNum = (obj && obj.nutNum) || 0
		this.washerNum = (obj && obj.washerNum) || 0
	}
}

export default ConstructionBolt
